OpenMarket/OpenMarket/Resource/Controller/ProductsViewController.swift

```diff
@@ -0,0 +1,207 @@
+
+        return segment
+    }()
+    
+    private lazy var collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        let collectionView = UICollectionView(frame: view.bounds, collectionViewLayout: layout)
+        
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        
+        collectionView.contentInset = UIEdgeInsets(
+            top: Constant.edgeInsetValue,
+            left: Constant.edgeInsetValue,
+            bottom: 0,
+            right: Constant.edgeInsetValue
+        )
+
+        collectionView.register(
+            ProductItemCell.self,
+            forCellWithReuseIdentifier: ProductItemCell.identifier
+        )
+        
+        return collectionView
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavbar()
+        view = collectionView
+        fetchData()
+    }
+}
+
+// MARK: UICollectionViewDelegateFlowLayout
+extension ProductsViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAt indexPath: IndexPath
+    ) -> CGSize {
+        let size = collectionView.bounds.size
+        let index = (segmentIndex + 1)
+        let contentsWidth = (size.width / CGFloat(index)) - (2 * Constant.edgeInsetValue)
+        let contentsHeight = index == 1 ? size.height * 0.1 : size.height * 0.3
+         
+        return CGSize(width: contentsWidth, height: contentsHeight)
+    }
+    
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        minimumLineSpacingForSectionAt section: Int
+    ) -> CGFloat {
+        return Constant.edgeInsetValue
+    }
+    
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        minimumInteritemSpacingForSectionAt section: Int
+    ) -> CGFloat {
+        return Constant.edgeInsetValue
+    }
+}
+
+// MARK: UICollectionViewDataSource
+extension ProductsViewController: UICollectionViewDataSource {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        numberOfItemsInSection section: Int
+    ) -> Int {
+        guard let products = productsData?.products else { return 0 }
+        return products.count
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        guard let products = productsData?.products else { return UICollectionViewCell() }
+        guard let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: ProductItemCell.identifier,
+            for: indexPath
+        ) as? ProductItemCell else {
+            return UICollectionViewCell()
+        }
+        let product = products[indexPath.row]
```

Comment:
```suggestion
        let product = products[safe :indexPath.row]
```

인덱스 이슈는 바로 앱이 죽는 현상으로 이어지기 대문에 방어코드가 있으면 좋을 것 같습니다